virtio-queue/src/desc/mod.rs

```diff
@@ -0,0 +1,61 @@
+//! Descriptor types for virtio queue.
+
+use vm_memory::{ByteValued, Le16, Le32, Le64};
+
+pub mod split;
+
+/// A virtio descriptor's layout constraints with C representation.
+/// This is a unified representation of the memory layout order
+/// for packed descriptors and split descriptors.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct RawDescriptor(Le64, Le32, Le16, Le16);
```

Comment:
Can we make this a field struct instead of a tuple struct?

Reply:
(A field struct is just more readable/understandable, that's my reasoning)

Reply:
Emm, I also think using a field struct is a good choice, but naming the field is so difficult. Do you have some advise?

Reply:
Same as C?

```rust
    /// Guest physical address of device specific data.
    pub addr: Le64,
    /// Length of device specific data.
    pub len: Le32,
    /// Index of descriptor in the descriptor table.
    pub id: Le16,
    /// Includes next, write, and indirect bits.
    pub flags: Le16,
```

Reply:
These names are used for the packed virtqueue descriptor. For the split virtqueue descriptor, it is
```
struct virtq_desc { 
        /* Address (guest-physical). */ 
        le64 addr; 
        /* Length. */ 
        le32 len; 
 
/* This marks a buffer as continuing via the next field. */ 
#define VIRTQ_DESC_F_NEXT   1 
/* This marks a buffer as device write-only (otherwise device read-only). */ 
#define VIRTQ_DESC_F_WRITE     2 
/* This means the buffer contains a list of buffer descriptors. */ 
#define VIRTQ_DESC_F_INDIRECT   4 
        /* The flags as indicated above. */ 
        le16 flags; 
        /* Next field if flags & NEXT */ 
        le16 next; 
};
```
And the RawDescriptor is an abstraction of these two. So I don't think it is suitable to use the packed descriptor's field name.

Reply:
Sorry, I copy pasted them in a hurry and forgot to say "like in the split `Descriptor` struct". So do the same struct definition but with the field names from C.

Reply:
> And the RawDescriptor is an abstraction of these two. So I don't think it is suitable to use the packed descriptor's field name.

Yeah, putting a name is difficult and using the `split` one could be confusing. I'd stay with tuple struct. In theory, the user should not touch it, right?